pastis/launchers/run_luvoir_efields.py

```diff
@@ -1,58 +1,71 @@
-    # DM_SPEC = (fpath, pad_orientations, True, False, False)
+    # If using Harris deformable mirror
+    if WHICH_DM == 'harris_seg_mirror':
+        fpath = CONFIG_PASTIS.get('LUVOIR', 'harris_data_path')  # path to Harris spreadsheet
+        pad_orientations = np.pi / 2 * np.ones(CONFIG_PASTIS.getint('LUVOIR', 'nb_subapertures'))
+        DM_SPEC = (fpath, pad_orientations, True, False, False)
 
-    # If Segmented Zernike Mirror, uncomment the following lines
-    DM = 'seg_mirror'  # Possible: "seg_mirror", "harris_seg_mirror", "zernike_mirror"
-    DM_SPEC = 5
+    # If using Segmented Zernike Mirror
+    if WHICH_DM == 'seg_mirror':
+        DM_SPEC = 3
 
     APLC_DESIGN = 'small'
     # First generate a couple of matrices
-    run_matrix = MatrixEfieldLuvoirA(which_dm=DM, dm_spec=DM_SPEC, design=APLC_DESIGN,
+    run_matrix = MatrixEfieldLuvoirA(which_dm=WHICH_DM, dm_spec=DM_SPEC, design=APLC_DESIGN,
                                      calc_science=True, calc_wfs=True,
                                      initial_path=CONFIG_PASTIS.get('local', 'local_data_path'),
                                      norm_one_photon=True)
     run_matrix.calc()
     dir_run = run_matrix.overall_dir
     print(f'All saved to {dir_run}.')
 
-    # Calculate and save per segment static tolerance
+    # get the automatically saved pastis_matrix
     pastis_matrix = fits.getdata(os.path.join(dir_run, 'matrix_numerical', 'pastis_matrix.fits'))
 
     # get the unaberrated coro_psf after the matrix run
     e0_psf = fits.getdata(os.path.join(dir_run, 'unaberrated_coro_psf.fits'))  # already normalized to max of direct pdf
     dh_mask = np.array(run_matrix.simulator.dh_mask.shaped)
     contrast_floor = util.dh_mean(e0_psf, dh_mask)
 
-    # set the target contrast
+    # Calculate the static tolerances
     c_target = 5.3*1e-11
     mus = calculate_segment_constraints(pastis_matrix, c_target=c_target, coronagraph_floor=contrast_floor)
-    np.savetxt(os.path.join(dir_run, 'mu_map_%s.csv' % c_target), mus, delimiter=',')
+    np.savetxt(os.path.join(dir_run, f'mu_map_{c_target:.2e}.csv'), mus, delimiter=',')
 
     num_modes = 5  # for harris thermal map or number of localized zernike modes
     nseg = run_matrix.simulator.nseg
 
-    ppl.plot_thermal_mus(mus, num_modes, nseg, c_target, out_dir=dir_run, save=True)
+    coeffs_table = util.sort_1d_mus_per_segment(mus, num_modes, nseg)
+    mu_list = []
+    label_list = []
+    for i in range(coeffs_table.shape[0]):
+        mu_list.append(coeffs_table[i])
+        if WHICH_DM == 'seg_mirror':
+            label_list.append(f'Zernike mode {i}')
+    if WHICH_DM == 'harris_seg_mirror':
+        label_list = ['Faceplates Silvered', 'Bulk', 'Gradient Radial', 'Gradient X lateral', 'Gradient Z axial']
 
+    ppl.plot_segment_weights(mu_list, dir_run, c_target, labels=label_list, fname=f'stat_1d_mus_{c_target:.2e}', save=True)
```

Comment:
Can you please add plots generated by this function?

Reply:
Adding some plots generated by this function.
[stat_1d_mus_1.00e-10.pdf](https://github.com/spacetelescope/PASTIS/files/9872527/stat_1d_mus_1.00e-10.pdf)

However this doesn't work for piston aberration, I get the following message:
<img width="996" alt="Screen Shot 2022-10-26 at 2 06 33 PM" src="https://user-images.githubusercontent.com/33359812/198103543-25ffa1fb-485f-4421-b37a-16d433c97c05.png">

Reply:
Nice catch, I'll fix this.

Reply:
Fixed now.